### PR TITLE
chore(kayenta): use new API for synthetic stages

### PR DESCRIPTION
**Pull request by robfletcher**

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/cluster/RollbackClusterStage.java [hunk 1]
```diff
@@ -89,21 +91,19 @@ public void afterStages(@Nonnull Stage parent, @Nonnull StageGraphBuilder graph)
       // propagate any attributes of the parent stage that are relevant to this rollback
       context.putAll(propagateParentStageContext(parent.getParent()));
 
-      Stage rollbackStage = graph.add((it) -> {
+      graph.append((it) -> {
```
Comment (robfletcher) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/cluster/RollbackClusterStage.java]: This was incorrect. Using `add` would create a stage with no `requisiteStageRefIds` when what's intended here is to run each region in serial. Spotted it when updating stages to use the new `append` method.

Comment (ajordens) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/cluster/RollbackClusterStage.java]: 👍

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/cluster/RollbackClusterStage.java [hunk 2]
```diff
@@ -49,13 +49,15 @@ public RollbackClusterStage(RollbackServerGroupStage rollbackServerGroupStage,
   }
 
   @Override
-  public void taskGraph(@Nonnull Stage stage, @Nonnull TaskNode.Builder builder) {
+  public void taskGraph(
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/cluster/RollbackClusterStage.java]: mr line breaky over here!

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/graph/StageGraphBuilder.java [hunk 3]
```diff
@@ -132,6 +134,28 @@ public void connect(@Nonnull Stage previous, @Nonnull Stage next) {
...
+   */
+  public @Nonnull Stage append(
+    @Nonnull Consumer<Stage> init
+  ) {
+    if (lastAdded == null) {
+      return add(init);
+    } else {
+      return connect(lastAdded, init);
+    }
+  }
```
Comment (robzienert) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/graph/StageGraphBuilder.java]: `append` is a great addition. Will update my other PR to use this instead.

Comment (robfletcher) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/graph/StageGraphBuilder.java]: Yeah, I found myself doing exactly that logic in various stage builders. Now you only need to keep variables of stages if you want to do something more complex like a fork/join.

**Pull request merged**